pkg/mimir/modules.go

```diff
@@ -1024,6 +1031,54 @@ func (t *Mimir) initUsageStats() (services.Service, error) {
 	return t.UsageStatsReporter, nil
 }
 
+func (t *Mimir) initContinuousTest() (services.Service, error) {
+	start := func(ctx context.Context) error {
+		logger := util_log.Logger
+		var err error
+
+		level.Info(logger).Log("msg", "initing continuous-test")
+		// Setting the environment variable JAEGER_AGENT_HOST enables tracing.
+		if trace, err := tracing.NewFromEnv("mimir-continuous-test", jaegercfg.MaxTagValueLength(16e3)); err != nil {
+			level.Error(logger).Log("msg", "Failed to setup tracing", "err", err.Error())
+		} else {
+			defer trace.Close()
+		}
+		level.Info(logger).Log("msg", "continuous-test setup tracing happened")
+
+		// Run the instrumentation server.
+		registry := prometheus.NewRegistry()
+		registry.MustRegister(version.NewCollector("mimir_continuous_test"))
+		registry.MustRegister(collectors.NewGoCollector())
+		level.Info(logger).Log("msg", "prom registry created for ct")
+
+		i := instrumentation.NewMetricsServer(t.Cfg.ContinuousTest.ServerMetricsPort, registry)
+
+		level.Info(logger).Log("msg", "metrics server created for ct")
+		if err := i.Start(); err != nil {
+			level.Error(logger).Log("msg", "Unable to start instrumentation server", "err", err.Error())
+			util_log.Flush()
+		}
+
+		// Init the client used to write/read to/from Mimir.
+		// TODO: building an external client and then going back in on a continuous test that's a module in Mimir feels.... ick. Clean this up
+		client, err := continuoustest.NewClient(t.Cfg.ContinuousTest.Client, logger)
+		level.Info(logger).Log("msg", "new client created for ct")
+		if err != nil {
+			level.Error(logger).Log("msg", "Failed to initialize client", "err", err.Error())
+			util_log.Flush()
```

Comment:
You should remove `util_log.Flush()` here, and you also also return an error from this function instead of `level.Error()`, so:

```suggestion
			return errors.Wrap(err, "failed to initialize continuous-test client")
```